tests/blotter/test_core.py

```diff
@@ -0,0 +1,173 @@
+from zipline.finance.blotter import SimulatedBlotter
+from zipline.finance.blotter.blotter import Blotter
+from zipline.finance.blotter.core import _make_blotters_core
+from zipline.testing.fixtures import ZiplineTestCase
+from zipline.testing.predicates import (
+    assert_equal,
+    assert_is,
+    assert_raises_str,
+)
+from zipline.utils.compat import mappingproxy
+
+
+class BlotterCoreTestCase(ZiplineTestCase):
+
+    def init_instance_fixtures(self):
+        super(BlotterCoreTestCase, self).init_instance_fixtures()
+
+        self.blotters, self.register, self.unregister, self.load = (
+            _make_blotters_core()
+        )
+
+        # make sure this starts empty
+        assert_equal(self.blotters, mappingproxy({}))
+
+    def test_load_not_registered(self):
+        msg = "no blotter class registered as 'ayy-lmao', options are: []"
+        with assert_raises_str(ValueError, msg):
+            self.load('ayy-lmao')
+
+        # register in reverse order to test the sorting of the options
+        self.register('c', SimulatedBlotter)
+        self.register('b', SimulatedBlotter)
+        self.register('a', SimulatedBlotter)
+
+        msg = (
+            "no blotter class registered as 'ayy-lmao', options are: "
+            "['a', 'b', 'c']"
+        )
+        with assert_raises_str(ValueError, msg):
+            self.load('ayy-lmao')
+
+    def test_register_decorator(self):
+
+        @self.register('ayy-lmao')
+        class ProperDummyBlotter(Blotter):
+
+            def order(self, asset, amount, style, order_id=None):
+                pass
+
+            def cancel(self, order_id, relay_status=True):
+                pass
+
+            def cancel_all_orders_for_asset(self, asset, warn=False,
+                                            relay_status=True):
+                pass
+
+            def execute_cancel_policy(self, event):
+                pass
+
+            def reject(self, order_id, reason=''):
+                pass
+
+            def hold(self, order_id, reason=''):
+                pass
+
+            def process_splits(self, splits):
+                pass
+
+            def get_transactions(self, bar_data):
+                pass
+
+            def prune_orders(self, closed_orders):
+                pass
+
+        expected_blotters = mappingproxy({'ayy-lmao': ProperDummyBlotter})
+        assert_equal(self.blotters, expected_blotters)
+        assert_is(self.load('ayy-lmao'), ProperDummyBlotter)
+
+        msg = "blotter class 'ayy-lmao' is already registered"
+        with assert_raises_str(ValueError, msg):
+            @self.register('ayy-lmao')
+            class Fake(object):
+                pass
+
+        msg = "The class specified is not a subclass of Blotter"
+        with assert_raises_str(TypeError, msg):
+            @self.register('something-different')
+            class ImproperDummyBlotter(object):
```

Comment:
why does this raise `TypeError` but `Fake` raised a `ValueError`

Reply:
It's testing 2 different types of issues that can come up when registering blotter classes - namely one is testing a class getting registered with a duplicate name, while the other is testing the registration of a class that isn't a subclass of Blotter. I can change this, or just rename the classes to make it more obvious - let me know what works better.

Reply:
Oh, I misread this and thought the error messages were the same. that's my bad.